### ISSUE TITLE
[Accessibility] div.monaco-tokenized-source requires aria-live="assertive"

### ISSUE THREAD

**Issue by jooyoungseo** [microsoft/vscode#167348]
CC @isidorn and @meganrogge

## Current Issue

1. Load NVDA or any screen reader.

1. Launch VSCode.

1. enable copilot extension.

1. Create any ts or js file.

1. Try to trigger copilot-powered inline suggestion (`Alt+\`).

1. See (hear) if suggestion pop-up is automatically read: it's not unless you press `Ctrl+K, Ctrl+I` to explicitly move your focus to the tooltip area.

* Notes: This is because `"div.monaco-tokenized-source"` does not have `aria-live="assertive"` to announce the dymanic content to screen reader.

* Notes: copilot is just one of the examples. Any extensions or widgets that depend `"div.monaco-tokenized-source"` have the same issue.

## Expected Behavior

The suggested dynamic content needs to be automatically announced to screen reader users.

* Note: I am willing to fix this issue and test with my screen reader.

## System Info

Version: 1.74.0-insider (system setup)
Commit: d0e44be43287acadb13f7c2808332e00fe52568d
Date: 2022-11-25T08:00:08.922Z
Electron: 19.1.6
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.22621
Sandboxed: Yes

**Comment by isidorn** [comment]
@jooyoungseo thanks a lot for opening this issue and for the PR.
However I am not sure if this is the ideal experience. I worry that it might result in Screen Reader reading too much, due to inline completions popping up all the time.
fyi @hediet since we should think about inline completion accessibility improvements.

**Comment by jooyoungseo** [comment]
@isidorn -- I understand your concern. However, adding aria-live to `div.monaco-tokenized-source` only reads new suggested token instead of the whole tooltip content. This means screen reader users do not necessarily hear redundant and verbose tooltip instructions and button names all the time. Without this feature, using inline suggestion is quite tedious for screen reader users. Because you need to manually press `Ctrl+K, Ctrl+I` each every time to identify the new content in tooltip.

As far as I tested, this gave me a more seamless and accessible experience when using inline suggestions. However, I understand that I cannot represent other screen reader users.

As a fair compromise, I suggest adding an option to VSCode settings to adjust the `aria-live` level between "on" and "off" so that users can customize the inline suggestion announcement level per their preference.

**Comment by meganrogge** [comment]
I do not have copilot. @hediet could you pls try out the PR with a screen reader?

**Comment by jooyoungseo** [comment]
I will close my PR, but keep opening this issue because I receive too many GH action messages. However, this issue still needs attention.

**Comment by hediet** [comment]
The idea was to use audio cues to indicate that inline suggestions are available.
Then, Ctrl+K, Ctrl+I can be pressed to read the suggestion.

The problem is that inline suggestions might be there 50% of the time you type something.

**Comment by meganrogge** [comment]
I think it makes sense to let the user decide if the potential disruption is worth it and disable this by default.

**Comment by meganrogge** [comment]
I explored reading the hint automatically using aria-live assertive. It was very disruptive because would suggest something with every keystroke. I also tried debouncing and the experience still was not great. You can try my branch merogge/ariaAlert to see what you think. 

Suggestion from @jooyoungseo : 

We can consider replacing the “Suggest  List box” message with a specific audio cue like an inline hint.

**Comment by jooyoungseo** [comment]
@meganrogge-- I have just tested your branch and this is exactly what I intended and proposed. You may find `aria-live` very disruptive; however, this is what sighted people see visually and it is the matter of getting used to. Advanced screen reader users have very sophisticated techniques. For example, screen readers have a typing echo setting to choose whether you will hear the typing feedback based on character/word/none/both. I would turn off the typing echo so that I can only hear the suggestion live event while typing. You can simply keep typing if you don't want to what's suggested and you can hit tab/enter key to accept the suggestion. Some advanced users, including myself, would find this feature useful and I believe this can improve their productivity. That's why I am suggesting this feature as a user configuration. I would definitely turn this on because the auto-suggestion and copilot hint is immediately read aloud which saves me a lot of time!

**Comment by meganrogge** [comment]
Great, thanks for resting. I think it makes a lot of sense to have a setting which enables this.

**Comment by meganrogge** [comment]
Going to investigate adding a setting for this and not having a dialogue box open up. 

`alert` does not work on macOS in any of the contexts for me, but I'm going to test on Windows

**Comment by meganrogge** [comment]
I have reached out to an apple contact about VoiceOver. This is fixed on Windows though

**Comment by meganrogge** [comment]
This now works on all platforms. I had an issue with my VoiceOver configuration
